src/mem.rs

```diff
@@ -1,377 +1,348 @@
-//! Reference implementation of the store traits.
+#![allow(missing_docs)]
 use crate::block::Block;
 use crate::cid::Cid;
 use crate::codec::Decode;
 use crate::error::{BlockNotFound, Result};
 use crate::ipld::Ipld;
-use crate::store::{AliasStore, Op, Store, StoreParams, Transaction};
-use async_std::sync::{Arc, RwLock};
+use crate::store::{Store, StoreParams};
 use async_trait::async_trait;
-use std::borrow::Borrow;
-use std::collections::{HashMap, HashSet};
-use std::ops::Deref;
-
-/// The status of a block.
-#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
-pub struct Status {
-    pinned: u32,
-    referenced: u32,
-}
+use std::collections::{BTreeMap, HashMap, HashSet};
+use std::sync::{Arc, Mutex};
 
-impl Status {
-    /// Creates a new status.
-    pub fn new(pinned: u32, referenced: u32) -> Self {
-        Self { pinned, referenced }
-    }
+type Id = u64;
+type Atime = u64;
 
-    /// Returns the number of times the block is pinned.
-    pub fn pinned(&self) -> u32 {
-        self.pinned
-    }
+struct BlockStore<S: StoreParams> {
+    next_id: Id,
+    blocks: HashMap<Id, Block<S>>,
```

Comment:
Not sure how relevant this is, but if you care about performance here I would suggest switching to Fnv HashMap. It is the standard hashmap with the fnv hash algorithm. I did some benchmarks for another thing I am working on, and was surprised how big the difference is.

It has 0 dependencies, and you probably have it somewhere in your tree already.

https://crates.io/crates/fnv

![image](https://user-images.githubusercontent.com/248257/93851079-c6064f00-fcaf-11ea-87cf-a2579f25f972.png)

Reply:
Interesting. The memstore is really intended for testing or as a reference for people implementing a store. but since it's such a small dependency I made the change.